craft_application/services/init.py

```diff
@@ -0,0 +1,218 @@
+                    "would overwrite existing files.\nExisting files are:\n  - "
+                    f"{existing_files_formatted}"
+                ),
+                resolution=(
+                    "Initialise the project in an empty directory or remove the existing files."
+                ),
+                retcode=os.EX_CANTCREAT,
+            )
+
+    def _copy_template_file(
+        self,
+        template_name: str,
+        template_dir: pathlib.Path,
+        project_dir: pathlib.Path,
+    ) -> None:
+        """Copy the non-ninja template from template_dir to project_dir.
+
+        If the file already exists in the projects copying is skipped.
+
+        :param project_dir: The directory to render the files into.
+        :param template_dir: The directory where templates are stored.
+        :param template_name: Name of the template to copy.
+        """
+        emit.debug(f"Copying file {template_name} to {project_dir}")
+        template_file = template_dir / template_name
+        destination_file = project_dir / template_name
+        if destination_file.exists():
+            emit.debug(f"Skipping file {template_name} because it is already present.")
+            return
+        destination_file.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(template_file, destination_file, follow_symlinks=False)
+
+    def _render_project(
+        self,
+        environment: jinja2.Environment,
+        project_dir: pathlib.Path,
+        template_dir: pathlib.Path,
+        context: dict[str, Any],
+    ) -> None:
+        """Render files for a project from a template.
+
+        :param environment: The Jinja environment to use.
+        :param project_dir: The directory to render the files into.
+        :param template_dir: The directory where templates are stored.
+        :param context: The context to render the templates with.
+        """
+        emit.progress("Rendering project.")
+        for template_name in environment.list_templates():
+            if not template_name.endswith(".j2"):
+                self._copy_template_file(template_name, template_dir, project_dir)
+                continue
+            template = environment.get_template(template_name)
+
+            # trim off `.j2`
+            rendered_template_name = template_name[:-3]
+            emit.debug(f"Rendering {template_name} to {rendered_template_name}")
+
+            path = project_dir / rendered_template_name
+            if path.exists():
+                emit.trace(f"Skipping file {template_name} as it is already present")
+                continue
+            path.parent.mkdir(parents=True, exist_ok=True)
+            with path.open("wt", encoding="utf8") as file:
+                out = template.render(context)
+                file.write(out)
+                if is_executable(template_dir / template_name) and os.name == "posix":
+                    make_executable(file)
+                    emit.debug("  made executable")
```

Comment:
Why not just always copy the mode?

```suggestion
                file.chmod((template_dir / template_name).stat().st_mode)
```

That way the template author can set the file exactly how they want. It's also a closer match to the behaviour of `shutil.copy2` for files that don't need rendering.

In fact if you want to make it even closer:


```suggestion
            shutil.copystat(template_dir / template_name, path)
```

Reply:
This is a great simplification, especially since the non-template files are already using `shutil.copy2()`.